src/Mediator.java

```diff
@@ -40,6 +40,15 @@ public <T> void sendInput(UserRequestProvider<T> provider) {
 		if(provider == null) {
 			throw new IllegalArgumentException("UserRequestProvider cannot be null");
 		}
+		try {
+			provider.propigateResponse(sendInputHelper(provider));
+		} catch (Exception e) {
```

Comment:
this as well you could do IOException e since its for input. at least i think it should work.